src/jvm.rs

```diff
@@ -178,6 +180,48 @@ impl<'jvm> Jvm<'jvm> {
     }
 }
 
+pub struct JvmBuilder<'args> {
+    args_builder: InitArgsBuilder<'args>,
+}
+
+impl<'args> JvmBuilder<'args> {
+    fn new() -> Self {
+        let args_builder = InitArgsBuilder::new()
+            .version(jni::JNIVersion::V8)
+            .option("-Xcheck:jni");
+        JvmBuilder {
+            args_builder
+        }
+    }
+
+    pub fn add_classpath(mut self, classpath: impl Display) -> Self {
+        self.args_builder = self.args_builder.option(format!("-Djava.class.path={classpath}"));
+        self
+    }
+
+    pub fn custom(mut self, opt_string: impl AsRef<str> + Into<Cow<'args, str>>) -> Self {
+        self.args_builder = self.args_builder.option(opt_string);
+        self
+    }
+
+    fn build_jvm(self) -> crate::GlobalResult<JavaVM> {
+        let jvm_args = self.args_builder.build()?;
+        let jvm = JavaVM::new(jvm_args)?;
```

Comment:
Rather than fail here if a JVM already exists, what if we just grabbed a pointer to it instead? This would allow duchess to be used as part of a library where consumers can "bring their own JVM".

Reply:
Good point. It would be useful for the case where `build_jvm` is called from `Jvm::with(..)`. In `get_or_default_init_jvm` it should be possible to catch the error result and then call [`JNI_GetCreatedJavaVMs`](https://docs.oracle.com/javase/8/docs/technotes/guides/jni/spec/invocation.html#JNI_GetCreatedJavaVMs) to obtain the existing JVM. However, as far as I know `JNI_GetCreatedJavaVMs` is not exposed by `jni`. Instead of starting to use `jni_sys` in this PR, it might be better to wait until your planned refactoring.

When calling `build_jvm` from `Jvm::builder().set_classpath(..).launch(..)`, the failure is intended because the consumer explicitly asked for a JVM with a certain classpath. Does that sound good?